torch/_functorch/vmap.py

```diff
@@ -391,19 +398,114 @@ def vmap(
         sequences out of the box.
     """
     _check_randomness_arg(randomness)
+    if not (chunk_size is None or chunk_size > 0):
+        raise ValueError("vmap: `chunk_size` should be greater than 0.")
+
+    def _get_chunked_inputs(flat_args, flat_in_dims, batch_size, chunk_size):
```

Comment:
nit: Is it possible to split this off into a helper function that is not declared inside of `vmap`? It makes the implementation look a bit nicer